src/backend/cuda/subconfigure.m4

```diff
@@ -61,25 +61,24 @@ if test "$with_cuda" != "no" ; then
     PAC_CHECK_HEADER_LIB([cuda_runtime_api.h],[cudart],[cudaStreamSynchronize],[have_cuda=yes],[have_cuda=no])
     if test "${have_cuda}" = "yes" ; then
         AC_MSG_CHECKING([whether nvcc works])
-        cat>conftest.cu<<EOF
-        __global__ void foo(int x) {}
-EOF
         if test -n "$ac_save_CC" ; then
-            NVCC_FLAGS="-ccbin $ac_save_CC"
+            NVCC_FLAGS="$NVCC_FLAGS -ccbin $ac_save_CC"
             # - pgcc/nvc doesn't work, use pgc++/nvc++ instead
             # - Extra optins such as `gcc -std=gnu99` doesn't work, strip the option
             NVCC_FLAGS=$(echo $NVCC_FLAGS | sed -e 's/nvc/nvc++/g' -e 's/pgcc/pgc++/g' -e's/ -std=.*//g')
-        else
-            NVCC_FLAGS=""
         fi
         # try nvcc from PATH if 'with-cuda' does not contain a valid path
         if test -d ${with_cuda} ; then
             nvcc_bin=${with_cuda}/bin/nvcc
         else
             nvcc_bin=nvcc
         fi
-        ${nvcc_bin} $NVCC_FLAGS -c conftest.cu 2> /dev/null
-        if test "$?" = "0" ; then
+        AC_LANG_PUSH([C])
+        ac_ext=cu
+        ac_compile='$nvcc_bin $NVCC_FLAGS -c conftest.$ac_ext >&5'
+        AC_COMPILE_IFELSE([AC_LANG_PROGRAM([__global__ void foo(int x) {}],[])],
+        [
+            AC_LANG_POP([C])
```

Comment:
Unbalanced PUSH/POP. The POP will be missed in the failure case, right?

Reply:
Yes, but the failure case calls `AC_MSG_ERROR`.

Reply:
I can put the AC_LANG_POP([C]) in the failure case anyway. It won't do any harm.

Reply:
Why not move AC_LANG_POP outside `AC_COMPILE_IFELSE`?

Reply:
Oh, I see. Are we doing CXX here?

Reply:
That was my original thought, but in the success case we then call AC_LINK_IFELSE with a C program.

Reply:
Maybe that portion should be moved out, too. Let me take one more crack.

Reply:
I guess I don't understand why we need AC_LANG_PUSH([C]) in the first place. We only configured for C in yaksa, right?

Reply:
My idea was to save the current language settings, then customize `ac_compile` and `ac_ext` to work with CUDA since there is no builtin autoconf support for it. Then AC_LANG_POP to restore the previous settings. Perhaps there is a better way?

Reply:
Now I see. Definitely need a comment here to help the understanding. But I do agree that I should have seen that. :) 

How about move the libstdcpp test outside and after the AC_COMPILE_IFELSE test. Since the other branch is just AC_MSG_ERROR, we can avoid the nesting and simplify the script a bit, right?

Reply:
Will do. Noticed one more issue I'll address in a 3rd commit, too.

Reply:
Now 4 commits, since I split out the un-nesting part from the AC_COMPILE_IFELSE usage.